docs/gadget-devel/metadata: Improve description of datasources (#3316)
inspektor-gadget/inspektor-gadget: docs/gadget-devel/metadata.md

### Datasources

The `datasources` section in the metadata file defines the datasources that the
gadget will use and their configuration. Each datasource can have `annotations`
that control different aspects of the datasource, as well as `fields` that can
also have `annotations` to control how they are formatted.

For example, the following metadata file defines a `foo` datasource with two
fields, `pid` and `bar`. In this case, both the datasource and the fields have
annotations:

```yaml
datasources:
  foo:
    annotations:
      cli.clear-screen-before: "true"
    fields:
      pid:
        annotations:
          description: Process ID
          template: pid
      bar:
        annotations:
          description: Bar field description
          columns.hidden: true
```

#### Annotations

##### Datasource

The annotations for the datasource depend on its type and the operators it uses.
For more information, refer to the [operator documentation](../reference/operators/).

##### Fields

> The fields annotations can be defined for each field of the datasource. The
> following annotations is a non-exhaustive list of the available annotations:

Reviewer: ```suggestion
The field annotations can be defined for each field of the datasource. The
following annotations are a non-exhaustive list of the available annotations:
```
